### PR TITLE
Improve Aquacell documentation

### PR DESCRIPTION
**Pull request by Jordi1990**
## Proposed change

Overall improvements to align with quality scale guidelines.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Updated AquaCell integration documentation with enhanced descriptions and hyperlinks.
	- Added a "Supported devices" section specifying compatibility with softener models.
	- Expanded prerequisites for integration setup with the official app.
	- Introduced configuration details for user login requirements.
	- Modified sensors section terminology for consistency.
	- Added "Use cases" and "Examples" sections for practical application guidance.
	- Clarified data polling frequency and confirmed no known limitations or issues.

### CODE REVIEW

File: source/_integrations/aquacell.markdown [hunk 1]
```diff
@@ -14,26 +14,90 @@ ha_platforms:
 ha_integration_type: device
 ---
 
-AquaCell is a water-softening device. The **AquaCell** {% term integration %} allows you to monitor your AquaCell device in Home Assistant.
+AquaCell is a water-softening device. The [AquaCell](https://www.aquacell-waterontharder.nl/) {% term integration %} allows you to monitor your AquaCell device in Home Assistant.
 You will need your Aquacell account information as used in the **AquaCell** app.
 
-This integration also supports **Harvey** softeners.
+This integration also supports [Harvey](https://www.harveywatersofteners.co.uk/) softeners.
```
Comment (frenck) [inline source/_integrations/aquacell.markdown:20]: Should this be added as an virtual integration instead?

File: source/_integrations/aquacell.markdown [hunk 2]
```diff
@@ -14,26 +14,90 @@ ha_platforms:
...
 - Wi-Fi signal strength.
 
-<div class="note">
-The device does not update frequently, the integration polls the information once a day from the cloud.
-</div>
+## Use cases
+
+The integration provides sensors to monitor salt level of the softener. You can use this information to create automations, for example, to notify you when the salt level is low and a refill is needed.
+
+You can also easily plot the history of the salt level sensors over time, which can give you more insight on usage.
```
Comment (frenck) [inline source/_integrations/aquacell.markdown]:
```suggestion
The integration provides sensors to monitor the salt level of the softener. You can use this information to create automations, for example, to notify you when the salt level is low and a refill is needed.

You can also easily plot the history of the salt level sensors over time, which can give you more insight into usage.
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/aquacell.markdown (hunk 1 above)
Comment (Jordi1990) [reply source/_integrations/aquacell.markdown:20]: Yes, I plan to make a PR for that soon. But it seemed right to also mention it here.

**Review by frenck — APPROVED** [review]
Thanks, @Jordi1990 👍 

../Frenck

**Pull request merged**